torch/csrc/api/include/torch/mps.h

```diff
@@ -5,6 +5,15 @@
 #include <cstddef>
 #include <cstdint>
 
+#ifdef __OBJC__
+#include <Foundation/Foundation.h>
+#include <Metal/Metal.h>
+typedef id<MTLCommandBuffer> MTLCommandBuffer_t;
+#else
+typedef void* MTLCommandBuffer_t;
+typedef void* dispatch_queue_t;
+#endif
```

Comment:
Why not guard the entire header with `#ifdef __OBJC__` as it's clearly not usable unless OBJC is defined, is it?

Reply:
Only following APIs require ObjC, the rest are usable with C++:
torch::mps::get_command_buffer()
torch::mps::get_dispatch_queue()

Reply:
Sure, but my point is one can't really use ObjC Blocks or `MTLCommandBuffer_t` , so aliasing those to `void*` is kind of pointless.

Reply:
OK, I'll update the PR and will remove the typedefs.

Reply:
I updated the PR. I couldn't find any better way other than renaming the typedef of  `dispatch_queue_t` to `DispatchQueue_t`.

Two facts here:
1- We cannot guard the entire `mps.h` with `#ifdef __OBJC__`, since this header along with `mps.cpp` MUST build successfully on all platforms to be able to use APIs like `torch::mps::is_available()` on every platform for backend availability checking.
2- It's not possible to partially guard the code with `#ifdef __OBJC__` since mps.cpp is a CPP file and that condition will always be false. Also, we cannot rename it to mps.mm due to fact 1 above.

So, unless if you have a better suggestion that makes MPS's CPP APIs work everywhere, I think the solution I committed for `DispatchQueue_t` should be fine. Please let me know. Thanks.